src/components/PersonLink.tsx

```diff
@@ -0,0 +1,72 @@
+import { Link, useParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../types';
+
+interface Props {
+  person: Person
+  people: Person[]
+}
+
+function findPersonByName(
+  people: Person[],
+  name: string | null,
+): Person | null {
+  return people.find((person) => person.name === name) || null;
```

Comment:
no needs to return `null` since if you don't find user you needed you'll get falsy value anyway